tiledb/sm/array/array.cc

```diff
@@ -130,6 +130,9 @@ Status Array::open(
     metadata_.reset(timestamp_);
   }
 
+  // Compute non-empty domain
+  compute_non_empty_domain();
```

Comment:
Should this be lazy on first access to the non_empty_domain similar to array metadata?

Reply:
The non-empty domain is used internally during the reads anyway. Also this is not too costly to compute.